src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java

```diff
@@ -63,8 +64,16 @@
 	@PostMapping(headers = { "Accept=multipart/form-data" })
 	public ResponseEntity<String> create(@RequestPart(required = false) MultipartFile file,
 			@RequestPart ProductCatalogue productCatalogue) {
+
+		final String DOWNLOAD_PHOTO_END_POINT = "/multimedia/media/";
+
+		final String baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() +
+				DOWNLOAD_PHOTO_END_POINT;
+
 		try {
 			productCatalogueService.add(productCatalogue);
```

Comment:
We should change the add method so that the Id(pk) is injected during saving so that we dont have to invoke `productCatalogueService.getProductCatalogueByName` on line 75

Reply:
The return type of the add method in the repository is void. The object which is just persisted can not be retrieved through the add method. Moreover, the name of the product is unique therefore, it is called to get the unique id of the product to make the file URL first and also to persist the image file by creating multimediaDTO.